### ISSUE TITLE
MacOS - 10.15.7 - when focusing into Insomnia windows, app freezes for a couple of seconds

### ISSUE THREAD

**Issue by paveljanda** [getinsomnia/insomnia#2956]
**Describe the bug**
When having Insomnia open in background, many times when focusing into Insomnia windows, app freezes for a couple of seconds.

**Expected behavior**
Not to freeze.

**Desktop (please complete the following information):**
 - OS: MacOS - 10.15.7
 - Installation Method: Not sure
 - App Version 2020.5.2

**Comment by nijikokun** [comment]
Welcome, and sorry about the freezing. Need some additional input:

- How many endpoints have you created within Insomnia?
- How many workspaces do you have?
- What application, Core or Designer?
- Did this start with `2020.5.2`?

Can you also take a screenshot of your console logs after it freezes?

**Comment by paveljanda** [comment]
Hi.

1) Maybe 150
2) 3
3) Core
4) I am not really sure about it

After cleaning out console output and then noticing that behaviour again, there was no output.

But I have to say, it is really painful to work with :/

Thanks for looking into that

**Comment by paveljanda** [comment]
Any luck solving this issue? It is completely unusable in this stage :( Sorry

**Comment by paveljanda** [comment]
Hi, any progress here?

**Comment by dimitropoulos** [comment]
@paveljanda yes, we've been working on a few mac-specific things, and we also recently upgraded electron (which solved a few things).  are you able to try with https://github.com/Kong/insomnia/releases/tag/core%402021.2.0

**Comment by paveljanda** [comment]
So far it looks good :) Great job! Thanks.

**Comment by paveljanda** [comment]
Sadly, I have to reopen this issue after further testing. The problem now occurs even if I ma still in the app. For example when focusing from JSON content window to adderess bar and back.

I tried reducing saved endpoints from 150 to maybe 30. That did not help.

**Comment by dimitropoulos** [comment]
What's your OS version (if changed since this issue was first created) and installation method?

**Comment by paveljanda** [comment]
OS version did not change, installation method: through https://insomnia.rest/download

**Comment by reynolek** [comment]
Hi @paveljanda -- Would it be possible for you to capture a screen recording of this occurring (along with the console open)? Might help us to start understanding what you are seeing, and hopefully help us identify a path to a fix.

**Comment by paveljanda** [comment]
https://user-images.githubusercontent.com/1488874/117434789-65b9da00-af2d-11eb-9a59-eb883cc3a770.mp4

The glitch start at cca 0:20.

It does not happen only on modal windows event, it happens usually when switching from another app to Insomnia or when switching from endpoint to endpoint

**Comment by sallmin** [comment]
I have the same issue. It freezes randomly every 3rd or 4th time I open the application. Freezes for 5-10 seconds, MacOS rainbow spinner shows, then continues working. It's been like this for over two months now

**Comment by cig-jaystevenson** [comment]
Also having this issue. Tried the resolution [here](https://github.com/Kong/insomnia/issues/282) (clearing the Response db) but the issue persists. 

Running on MacOS Ventura 13.4 (22F66) but has appeared in all previous versions as far as I can remember.

**Comment by eyalyoli** [comment]
also same issue here, on Sonoma 14
